src/PowerShell/PowerShellManager.cs

```diff
@@ -106,6 +123,27 @@ private void RegisterStreamEvents()
             _pwsh.Streams.Warning.DataAdding += streamHandler.WarningDataAdding;
         }
 
+        /// <summary>
+        /// Create the PowerShell function that is equivalent to the 'scriptFile' when possible.
+        /// </summary>
+        private void DeployAzFunctionToRunspace()
```

Comment:
PS scripts often use $MyInvocation or $PSCommandPath to locate files based on the current script file location. Will this still work?

Reply:
Should we give the user a way to opt out of this optimization (for this and similar reasons)? One workaround I see now is to just insert `#Requires`, perhaps we could come up with something more explicit.